Response object interface doesn't match @types/express behavior, it's not really generic

**Describe the bug**

@types/express-serve-static-core has the following interface for Response object:

```
export type Send<ResBody = any, T = Response<ResBody>> = (body?: ResBody) => T;

export interface Response<
    ResBody = any,
    Locals extends Record<string, any> = Record<string, any>,
    StatusCode extends number = number
> extends http.ServerResponse,
        Express.Response {
    ...
    /**
     * Send a response.
     *
     * Examples:
     *
     *     res.send(new Buffer('wahoo'));
     *     res.send({ some: 'json' });
     *     res.send('<p>some html</p>');
     *     res.status(404).send('Sorry, cant find that');
     */
    send: Send<ResBody, this>;
```

I want to have my controller functions fully typed from request to response, so with Express and @types/express i can do something like the following:

```
type ResponseBody<T = unknown> = {
  data: T,
  meta: {
    success: boolean,
    message: string,
    count?: number
  }
}

export type GenericResponse<T> = Response<ResponseBody<T>>
```

which I can then use in my controller functions to get type safety when calling res.send():

<img width="574" alt="Captura de pantalla 2022-12-11 a la(s) 9 35 32" src="https://user-images.githubusercontent.com/29734364/206919377-9adaf39a-173b-4b7d-9920-238aa68b2fbb.png">

<img width="1680" alt="Captura de pantalla 2022-12-11 a la(s) 9 36 35" src="https://user-images.githubusercontent.com/29734364/206919427-ab92f0f8-2606-4333-a70d-b9e5aa4aface.png">

With tinyhttp, using the exact same GenericResponse type but importing Response from '@tinyhttp/app' I don't get the same type safety, res.send() ends up with the following signature:

```
(method) Response<ResponseBody<string>>.send(body: unknown): Response<any>
```

**To Reproduce**

Steps to reproduce the behavior:

1. Pass type argument to generic Response interface
2. Call res.send() with an argument of mismatched type relative to argument passed to Response interface

**Expected behavior**

res.send() should expect an argument with the same type as the type argument passed to the Response interface

**Versions**

- `node`: 18.12.1
- `@tinyhttp/app`: 2.0.25

**Additional context**

I'm sorry if this is not a bug, but I couldn't come to any conclusion as to why this should be different from @types/express

probably related, currently it's impossible to use tinyhttp with typescript

<details>
 ✘  ~/r/m/backend  tsc index.ts
node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/append.d.ts:3:50 - error TS2315: Type 'ServerResponse' is not generic.

3 export declare const append: <Response_1 extends Res<import("http").IncomingMessage> = Res<import("http").IncomingMessage>>(res: Response_1) => (field: string, value: string | number | string[]) => Response_1;
                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/append.d.ts:3:88 - error TS2315: Type 'ServerResponse' is not generic.

3 export declare const append: <Response_1 extends Res<import("http").IncomingMessage> = Res<import("http").IncomingMessage>>(res: Response_1) => (field: string, value: string | number | string[]) => Response_1;
                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/cookie.d.ts:4:82 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const setCookie: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1 & {
                                                                                   ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/cookie.d.ts:4:93 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const setCookie: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1 & {
                                                                                              ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/cookie.d.ts:9:84 - error TS2315: Type 'ServerResponse' is not generic.

9 export declare const clearCookie: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (name: string, options?: cookie.SerializeOptions) => Response_1;
                                                                                     ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/cookie.d.ts:9:95 - error TS2315: Type 'ServerResponse' is not generic.

9 export declare const clearCookie: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (name: string, options?: cookie.SerializeOptions) => Response_1;
                                                                                                ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/download.d.ts:8:81 - error TS2315: Type 'ServerResponse' is not generic.

8 export declare const download: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (path: string, filename?: string | Callback, options?: DownloadOptions | Callback, cb?: Callback) => Response_1;
                                                                                  ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/download.d.ts:8:92 - error TS2315: Type 'ServerResponse' is not generic.

8 export declare const download: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (path: string, filename?: string | Callback, options?: DownloadOptions | Callback, cb?: Callback) => Response_1;
                                                                                             ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/download.d.ts:9:54 - error TS2315: Type 'ServerResponse' is not generic.

9 export declare const attachment: <Response_1 extends Res<Req>>(res: Response_1) => (filename?: string) => Response_1;
                                                       ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/format.d.ts:12:87 - error TS2315: Type 'ServerResponse' is not generic.

12 export declare const formatResponse: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>, Next extends next = next>(req: Request_1, res: Response_1, next: Next) => (obj: FormatProps) => Response_1;
                                                                                         ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/format.d.ts:12:98 - error TS2315: Type 'ServerResponse' is not generic.

12 export declare const formatResponse: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>, Next extends next = next>(req: Request_1, res: Response_1, next: Next) => (obj: FormatProps) => Response_1;
                                                                                                    ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:3:53 - error TS2315: Type 'ServerResponse' is not generic.

3 export declare const setHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string | Record<string, string | number | string[]>, val?: string | number | readonly string[]) => Response_1;
                                                      ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:3:64 - error TS2315: Type 'ServerResponse' is not generic.

3 export declare const setHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string | Record<string, string | number | string[]>, val?: string | number | readonly string[]) => Response_1;
                                                                 ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:4:90 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const setLocationHeader: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (url: string) => Response_1;
                                                                                           ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:4:101 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const setLocationHeader: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>>(req: Request_1, res: Response_1) => (url: string) => Response_1;
                                                                                                      ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:5:61 - error TS2315: Type 'ServerResponse' is not generic.

5 export declare const getResponseHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string) => string | number | string[];
                                                              ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:5:72 - error TS2315: Type 'ServerResponse' is not generic.

5 export declare const getResponseHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string) => string | number | string[];
                                                                         ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:6:58 - error TS2315: Type 'ServerResponse' is not generic.

6 export declare const setLinksHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (links: {
                                                           ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:6:69 - error TS2315: Type 'ServerResponse' is not generic.

6 export declare const setLinksHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (links: {
                                                                      ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:9:57 - error TS2315: Type 'ServerResponse' is not generic.

9 export declare const setVaryHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string) => Response_1;
                                                          ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:9:68 - error TS2315: Type 'ServerResponse' is not generic.

9 export declare const setVaryHeader: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (field: string) => Response_1;
                                                                     ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:10:58 - error TS2315: Type 'ServerResponse' is not generic.

10 export declare const setContentType: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (type: string) => Response_1;
                                                            ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:10:69 - error TS2315: Type 'ServerResponse' is not generic.

10 export declare const setContentType: <Response_1 extends Res<Req> = Res<Req>>(res: Response_1) => (type: string) => Response_1;
                                                                       ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/redirect.d.ts:4:81 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const redirect: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>, Next extends next = next>(req: Request_1, res: Response_1, next: Next) => (url: string, status?: number) => Response_1;
                                                                                  ~~~~~~~~

node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/redirect.d.ts:4:92 - error TS2315: Type 'ServerResponse' is not generic.

4 export declare const redirect: <Request_1 extends Req = Req, Response_1 extends Res<Req> = Res<Req>, Next extends next = next>(req: Request_1, res: Response_1, next: Next) => (url: string, status?: number) => Response_1;
                                                                                             ~~~~~~~~


Found 25 errors in 6 files.

Errors  Files
     2  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/append.d.ts:3
     4  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/cookie.d.ts:4
     3  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/download.d.ts:8
     2  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/format.d.ts:12
    12  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/headers.d.ts:3
     2  node_modules/.pnpm/@tinyhttp+res@2.0.20/node_modules/@tinyhttp/res/dist/redirect.d.ts:4
</details>

--- a/packages/app/src/response.ts
+++ b/packages/app/src/response.ts
@@ -19,30 +19,30 @@ export const renderTemplate =
     return res
   }
 
-export interface Response<O = any> extends ServerResponse {
-  header(field: string | Record<string, unknown>, val?: string | any[]): Response
-  set(field: string | Record<string, unknown>, val?: string | any[]): Response
+export interface Response<O = any, B = any> extends ServerResponse {
+  header(field: string | Record<string, unknown>, val?: string | any[]): Response<O, B>
+  set(field: string | Record<string, unknown>, val?: string | any[]): Response<O, B>
   get(field: string): string | number | string[]
-  send(body: unknown): Response
-  sendFile(path: string, options?: ReadStreamOptions, cb?: (err?: any) => void): Response
-  json(body: unknown): Response
-  status(status: number): Response
-  sendStatus(statusCode: number): Response
+  send(body: B): Response<O, B>
+  sendFile(path: string, options?: ReadStreamOptions, cb?: (err?: any) => void): Response<O, B>
+  json(body: B): Response<O, B>
+  status(status: number): Response<O, B>
+  sendStatus(statusCode: number): Response<O, B>
   cookie(
     name: string,
     value: string | Record<string, unknown>,
     options?: SerializeOptions & Partial<{ signed: boolean }>
-  ): Response
-  clearCookie(name: string, options?: SerializeOptions): Response
-  location(url: string): Response
-  links(links: { [key: string]: string }): Response
-  render(file: string, data?: Record<string, any>, options?: TemplateEngineOptions<O>): Response
-  vary(field: string): Response
-  format(obj: FormatProps): Response
-  redirect(url: string, status?: number): Response
-  type(type: string): Response
-  download(path: string, filename: string, options?: DownloadOptions, cb?: (err?: any) => void): Response
-  attachment(filename?: string): Response
+  ): Response<O, B>
+  clearCookie(name: string, options?: SerializeOptions): Response<O, B>
+  location(url: string): Response<O, B>
+  links(links: { [key: string]: string }): Response<O, B>
+  render(file: string, data?: Record<string, any>, options?: TemplateEngineOptions<O>): Response<O, B>
+  vary(field: string): Response<O, B>
+  format(obj: FormatProps): Response<O, B>
+  redirect(url: string, status?: number): Response<O, B>
+  type(type: string): Response<O, B>
+  download(path: string, filename: string, options?: DownloadOptions, cb?: (err?: any) => void): Response<O, B>
+  attachment(filename?: string): Response<O, B>
   app?: App
   locals?: Record<string, any>
   /**
@@ -52,7 +52,7 @@ export interface Response<O = any> extends ServerResponse {
    *
    * @param obj Response object
    */
-  jsonp(obj: any): Response
+  jsonp(obj: any): Response<O, B>
 
-  append(field: string, value: any): Response
+  append(field: string, value: any): Response<O, B>
 }

